component/azstorage/azauthmsi.go

```diff
@@ -198,35 +203,53 @@ func (azmsi *azAuthBlobMSI) getCredential() interface{} {
 		log.Info("azAuthBlobMSI::getCredential : MSI Token over CLI retrieved %s (%d)", token.AccessToken, token.Expires())
 		// We are running in cli mode so token can not be refreshed, on expiry just get the new token
 		tc = azblob.NewTokenCredential(token.AccessToken, func(tc azblob.TokenCredential) time.Duration {
-			newToken, err := azmsi.fetchTokenFromCLI()
-			if err != nil {
-				log.Err("azAuthBlobMSI::getCredential : Failed to refresh token [%s]", err.Error())
-				return 0
-			}
+			for failCount := 0; failCount < 5; failCount++ {
+				newToken, err := azmsi.fetchTokenFromCLI()
+				if err != nil {
+					log.Err("azAuthBlobMSI::getCredential : Failed to refresh token attempt %d [%s]", failCount, err.Error())
+					failCount++
```

Comment:
Why is `failCount` being incremented here? It is already done by the for loop.

Reply:
fixed